Dockerfile.vcsim

```diff
@@ -1,4 +1,45 @@
+# Create a builder container
+FROM golang@sha256:f254180c5defa2653955e963fb0626e3d4fbbb162f7cff6490e94607d1d867ff AS build
+WORKDIR /go/src/app
+
+# Create appuser to isolate potential vulnerabilities
+# See https://stackoverflow.com/a/55757473/12429735
+ENV USER=appuser
+ENV UID=10001
+RUN adduser \
+    --disabled-password \
+    --gecos "" \
+    --home "/nonexistent" \
+    --shell "/sbin/nologin" \
+    --no-create-home \
+    --uid "${UID}" \
+    "${USER}"
+
+# Build
+RUN CGO_ENABLED=0 go get -u github.com/vmware/govmomi/vcsim
+
+###############################################################################
+# Final stage
 FROM scratch
-LABEL maintainer="fabio@vmware.com"
-COPY vcsim /
-ENTRYPOINT [ "/vcsim" ]
\ No newline at end of file
+
+# Allow container to use latest TLS certificates
+COPY --from=build /etc/ssl/certs/ca-certificates.crt /etc/ssl/certs/
+
+# Copy over appuser to run as non-root
+COPY --from=build /etc/passwd /etc/passwd
+COPY --from=build /etc/group /etc/group
+
+# Copy over the /tmp directory for golang/os.TmpDir
+COPY --chown=appuser --from=build /tmp /tmp
+
+# Run all commands as non-root
+USER appuser:appuser
+
+# Application runs on local port
+EXPOSE 8989
```

Comment:
QQ: do you think we should hard-code or make this one configurable, e.g. via ENV?

Reply:
If the goal is to test vcsim, then testing changing ports is a valid test.